site/content/en/docs/handbook/pushing.md

```diff
@@ -2,71 +2,103 @@
+|  [minikube ssh](http://localhost:1313/docs/handbook/pushing/#building-images-inside-of-minikube-using-ssh)   |   all	|    |  best 	|
 
-The add command will store the requested image to `$MINIKUBE_HOME/cache/images`, and load it into the VM's container runtime environment next time `minikube start` is called.
 
-To display images you have added to the cache:
+* note1 : the default container-runtime on minikube is 'docker'.
+* note2 : 'none' driver (bare metal) does not need pushing image to the cluster, as any image on your system is already available to the kuberentes.
+
+---
+
+## 1.Pushing directly to the in-cluster Docker daemon (docker-env)
+When using a container or VM driver (all drivers except none), you can reuse the Docker daemon inside minikube cluster.
+this means you don't have to build on your host machine and push the image into a docker registry. You can just build inside the same docker daemon as minikube which speeds up local experiments.
+
+To point your terminal to use the docker daemon inside minikube run this:
 
 ```shell
-minikube cache list
+eval $(minikube docker-env)
 ```
 
-This listing will not include the images which are built-in to minikube.
-
+now any 'docker' command you run in this current terminal will run against the docker inside minikube VM or Container.
+Try it:
 
 ```shell
-minikube cache delete <image name>
+docker ps
 ```
 
-For more information, see:
+now you 'build' against the docker inside minikube. which is instantly accessible to kubernetes cluster.
 
-* [Reference: cache command]({{< ref "/docs/commands/cache.md" >}})
+'''
+docker build -t myimage .
+'''
 
-You must be using minikube with the container runtime set to Docker. This is the default setting.
+Remember to turn off the `imagePullPolicy:Always` (use `imagePullPolicy:IfNotPresent` or `imagePullPolicy:Never`), as otherwise Kubernetes won't use images you built locally.
 
-# Pushing directly to the in-cluster Docker daemon
-When user a container or VM driver, it's really handy to reuse the Docker daemon inside minikube; as this means you don't have to build on your host machine and push the image into a docker registry - you can just build inside the same docker daemon as minikube which speeds up local experiments.
 
-To point your terminal to use the docker daemon inside minikube run this:
+Please remember evalulating the docker-env is only valid for the current terminal.
+and by closing the terminal, you will go back to using your own system's docker daemon.
+to verify your terminal is using minikuber's docker-env you can check the value of the environment MINIKUBE_ACTIVE_DOCKERD
+
+more information on [docker-env](https://minikube.sigs.k8s.io/docs/commands/docker-env/)
+
+---
+
+## 2.Push images using 'cache' command.
+
+From the host, you can push a Docker image directly to minikube. It will also be cached for all minikube clusters .
```

Comment:
1. This image will be cached and automatically pulled into all future minikube clusters created on the machine
2. There is an extra space before the period at the end